[logs] Add autotuning data

stashuk-olek opened the pull request:
Summary: Add autotuning logging data to scuba/chrome trace.

Test Plan:
```
TORCHINDUCTOR_MAX_AUTOTUNE=1 tlp buck run //scripts/sashko:compilation_sample
```

Open https://interncache-all.fbcdn.net/manifold/perfetto-artifacts/tree/ui/index.html#!/viewer?local_cache_key=00000000-0000-0000-92db-f23383ebf5b5, search for template_autotuning, see in metadata strides (see screenshot)

Differential Revision: D75457770




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

masnesral commented on torch/_dynamo/utils.py:
```diff
@@ -726,7 +726,12 @@ def _foo(...):
         compilation_time_metrics[key].append(time_spent_ns / 1e9)
         chromium_log.log_event_end(
-            event_name, end_ns, {}, start_ns, log_pt2_compile_event, compile_id
+            event_name,
```
This is functionally the same, right? We don't need to change it?

stashuk-olek replied:
Yeah, lint failed for it in CI of PR, I was applying auto-lint. but let me revert it again

masnesral commented on torch/_inductor/select_algorithm.py:
```diff
@@ -2190,10 +2190,17 @@ def make_benchmark_fn():
         def autotune(choices):
             log.debug("Starting autotuning")
+
+            autotune_data = {
```
Metadata is already a dict, so it seems like this could just be the metadata instead of putting it inside another dict?

stashuk-olek committed:
[logs] Add autotuning data (#154771)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/154771

Add autotuning logging data to scuba/chrome trace.

topic: not user facing

Test Plan:
```
TORCHINDUCTOR_MAX_AUTOTUNE=1 tlp buck run //scripts/sashko:compilation_sample
```

Open https://interncache-all.fbcdn.net/manifold/perfetto-artifacts/tree/ui/index.html#!/viewer?local_cache_key=00000000-0000-0000-0585-9b65cb8d0554, search for template_autotuning, see in metadata strides (see screenshot)

{F1978899244}

topic: not user facing

Reviewed By: PaulZhang12, jamesjwu, masnesral

Differential Revision: D75457770

stashuk-olek commented:
@pytorchbot merge
